### PR TITLE
Added note for _FILE convention

### LINKED ISSUE #17774: Reading Swarm managed secrets into a container environment variable doesn't work
### Is this a docs issue?

- [X] My issue is about the documentation content or website

### Type of issue

Information is incorrect

### Description

The page titled "How to use secrets in Docker Compose" states...

- The secrets attribute under each service defines the secrets you want to inject into the specific container.
- The top-level secrets section defines the variables db_password and db_root_password and provides the file that populates their values.
- The deployment of each container means Docker creates a temporary filesystem mount under /run/secrets/<secret_name> with their specific values.

Which works. However, in the advanced compose file example it shows...

```
services:
   db:
     image: mysql:latest
     volumes:
       - db_data:/var/lib/mysql
     environment:
       MYSQL_ROOT_PASSWORD_FILE: /run/secrets/db_root_password
       MYSQL_DATABASE: wordpress
       MYSQL_USER: wordpress
       MYSQL_PASSWORD_FILE: /run/secrets/db_password
     secrets:
       - db_root_password
       - db_password
```

Which implies that MYSQL_PASSWORD_FILE environment variable declaration would read the contents of the file  /run/secrets/db_password and result with an environment variable MYSQL_PASSWORD containing the contents of the file. However, after much testing, I cannot get the _FILE feature to work. Instead the environment variable MYSQL_PASSWORD_FILE contains the string "/run/secrets/db_root_password"

Either this is a bug and it should work, or the documentation is wrong or outdated.

### Location

https://docs.docker.com/compose/use-secrets/

### Suggestion

Either update the documentation to reflect how the feature works, or maybe this should result in a bug in Docker itself to implement the desired functionality.

### PR DESCRIPTION
**Pull request by Affan-7**
### Proposed changes

Added note for `_FILE` convention. This note clears the ambiguity.

Closes #17774

### CODE REVIEW

**Comment by Affan-7** [comment]
Hi @neersighted, can you please review it?

File: compose/use-secrets.md [hunk 1]
```diff
@@ -74,9 +74,13 @@ volumes:
 In the advanced example above:
 
 - The `secrets` attribute under each service defines the secrets you want to inject into the specific container.
-- The top-level secrets section defines the variables `db_password` and `db_root_password` and provides the file that populates their values.
+- The top-level `secrets` section defines the variables `db_password` and `db_root_password` and provides the `file` that populates their values.
 - The deployment of each container means Docker creates a temporary filesystem mount under `/run/secrets/<secret_name>` with their specific values.
 
+> **Note**
+>
+> The [_FILE](https://hub.docker.com/_/mysql) convention is specific to mysql image.
```
Comment (neersighted) [inline compose/use-secrets.md:82]: We should make it clear that this is specific to the `mysql` image used in this example, as e.g. https://hub.docker.com/_/postgres also supports this convention, as does much of DOI (cc @tianon).

Maybe something more like:
```suggestion
> The `_FILE` environment variables demonstrated here are a convention used by some images, including Docker Official Images like [mysql](https://hub.docker.com/_/mysql) and [postgres](https://hub.docker.com/_/postgres).
```

Comment (Affan-7) [reply compose/use-secrets.md]: Looks good to me.

**Review by tianon — APPROVED** [review]

**Review by neersighted — APPROVED** [review]

**Comment by aevesdocker** [comment]
Thank you @Affan-7 for creating the PR

And thank you @neersighted and @tianon for the review 🙏

**Review by aevesdocker — APPROVED** [review]

**Pull request merged**